pkg/domain/entities/engine_image.go

```diff
@@ -22,7 +22,7 @@ type ImageEngine interface {
 	Push(ctx context.Context, source string, destination string, opts ImagePushOptions) error
 	Remove(ctx context.Context, images []string, opts ImageRemoveOptions) (*ImageRemoveReport, []error)
 	Save(ctx context.Context, nameOrID string, tags []string, options ImageSaveOptions) error
-	Scp(ctx context.Context, src, dst string, parentFlags []string, quiet bool) error
+	Scp(ctx context.Context, src, dst string, parentFlags []string, quiet bool, sshMode string) error
```

Comment:
Could this use `ssh.EngineMode`? (I have a personal vendetta against UNIX and untyped strings.)

In general, push string parsing and handling of invalid values as close to the UI as possible (i.e. somewhere close to the CLI layer, or close to the HTTP request handler).